frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditTaskRecommendedItems/NewNameRecommendation.tsx

```diff
@@ -0,0 +1,70 @@
+import { useValidateBpmnTaskId } from '../../../../hooks/useValidateBpmnId';
+import { useBpmnContext } from '../../../../contexts/BpmnContext';
+import { StudioIconTextfield, StudioRecommendedNextAction } from '@studio/components';
+import { KeyVerticalIcon } from '@studio/icons';
+import type Modeling from 'bpmn-js/lib/features/modeling/Modeling';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface NewNameRecommendationProps {}
+
+export const NewNameRecommendation = ({}: NewNameRecommendationProps): React.ReactElement => {
+  const { bpmnDetails, setBpmnDetails, modelerRef } = useBpmnContext();
+  const modelerInstance = modelerRef.current;
+  const modeling: Modeling = modelerInstance.get('modeling');
+  const { t } = useTranslation();
+  const { validateBpmnTaskId } = useValidateBpmnTaskId();
+
+  const [newName, setNewName] = useState('');
+  const [newNameError, setNewNameError] = useState('');
+
+  const saveNewName = () => {
+    modeling.updateProperties(bpmnDetails.element, {
+      id: newName,
+    });
+    setBpmnDetails({
+      ...bpmnDetails,
+      id: newName,
+
+      metadata: {
+        ...bpmnDetails.metadata,
+        justAdded: false,
```

Comment:
I'm just thinking, but from my point of view, I think `justAdded` is quite a loose name. It's difficult to understand what is used for. 

Another thing I think is that adding this data to the metadata on the `BpmnDetails` might not be that reusable. Would this data be sent over the HTTP to the backend as well?🤔 

What about adding a Context which holds an array of information about the items that should show NextRecommendedAction? Then we can also use the same approach for NextRecommendedAction in other places within Studio where we do not have BpmnDetails. 😊

Reply:
> Another thing I think is that adding this data to the metadata on the BpmnDetails might not be that reusable. Would this data be sent over the HTTP to the backend as well?🤔

As far as i know, only the element property for this type is sent to the backend 🤔 

> What about adding a Context which holds an array of information about the items that should show NextRecommendedAction? Then we can also use the same approach for NextRecommendedAction in other places within Studio where we do not have BpmnDetails. 😊

I'm curious about this, if i understand your suggestion correctly it would be something like this?
```typescript
RecommendedActionStack = [
    { type: "ADDED_BPMN_TYPE",              id: "activity_AbCd1234" },
    { type: "POSSIBLE_UPDATE_TO_NEW_TYPE",  id: "formItem_123" }
]
 ```
And each component using this context would fetch relevant entries from this array? I'm down to talk about this in a huddle if you want to discuss more in detail.

Reply:
Yes, I was thinking about something like that. Do we need an entire object with type and id? I do think we need to care about the order of the items, I believe it's easier to check if the item exists within the object. If it does exits then show the next recommended action.


### The Context
```typescript
// I made it an object here to enable a hashmap lookup instead of looping through an array. This is more efficient/performant.
const actions: { [key: string]: string } = { "activity_AbCd1234" , "formItem_123"};

const shouldDisplayRecAction = (actionId: string): boolean => {
   return actions[actionId];
}

const removeAction = (actionId: string): void => {
   delete actions[actionId];
}
```


### Component that might show next recommended action
```tsx
// Within the component that might show NextRecommendedAction

const { shouldDisplayNextRecAction, removeAction } = useNextRecommendedActionContext();

// This data will be dynamic based on props/data.
const componentId = "formItem_123";

const onRecommendedActionClosed = (): void => {
   removeAction(componentId);
}

if (shouldDisplayNextRecAction(componentId) {
  return <NextRecommendedAction onSkip={onRecommendedActionClosed} onSave={onRecommendedActionClosed }/>
}

return (<div>The other code to render</div>);
```


I’m happy to join a pair-programming session to discuss this further. The code above needs adjustments before it will compile and run, but it was written just to explain my thoughts.